src/middlewared/middlewared/job.py

```diff
@@ -130,13 +134,17 @@ def add(self, job):
                 queued_jobs = [another_job for another_job in self.queue if another_job.lock is job.lock]
                 if len(queued_jobs) >= job.options["lock_queue_size"]:
                     for queued_job in reversed(queued_jobs):
-                        if not credential_is_limited_to_own_jobs(job.credentials):
-                            return queued_job
                         if (
-                            job.credentials.is_user_session and
-                            queued_job.credentials.is_user_session and
-                            job.credentials.user['username'] == queued_job.credentials.user['username']
+                                not credential_is_limited_to_own_jobs(job.credentials) or (
+                                    job.credentials.is_user_session and
+                                    queued_job.credentials.is_user_session and
+                                    job.credentials.user['username'] == queued_job.credentials.user['username']
+                                )
                         ):
+                            if job.message_ids:
+                                queued_job.message_ids += job.message_ids
+                                queued_job.send_changed_event()
```

Comment:
Perhaps we could add a comment here to explain what's going on.

It _seems_ like whenever a job is created we send an event to all older instances of the same job, if we've exceeded the queue size.  Not sure I'm following the behavior.  Do we have it documented anywhere?  I don't notice it in _jobs.rst_ below.

Reply:
From the client's perspective it's documented in jobs.rst line 11, 12.

`lock` and `lock_queue_size` job parameters are documented in the `@job` decorator description (and, as a consequence, in the middleware developer documentation)